ProjectSidewalk/SidewalkWebpage
# Fixes first context menu disconnect

Resolves #3166 

Fixes the first context menu (either first label placed or first existing label clicked) being disconnected from the label and connector. Also redesigns code for determining if and where the context menu is displayed to replace hardcoded values with variables.

##### Before/After screenshots (if applicable)
Before: 
1. First label placed (or first existing label clicked)
![image](https://user-images.githubusercontent.com/106000281/234414683-21e5f657-bda1-4401-9ef1-17d8d004f990.png)

2. Close and reopen context menu (or place another label) 
![image](https://user-images.githubusercontent.com/106000281/234415087-776a3432-1a3b-4952-aa7c-7fddb3278bdf.png)

- Pedestrian Signal example
![image](https://user-images.githubusercontent.com/106000281/234415540-153c76b5-514d-4cf8-a860-434f1b0a37c2.png)

After:
1. First label placed (or first existing label clicked) 
![image](https://user-images.githubusercontent.com/106000281/234415936-e774d649-27e2-494f-a52a-e872d2e0b7f9.png)

2. Close and reopen context menu (or place another label) 
![image](https://user-images.githubusercontent.com/106000281/234416046-73216d36-65ca-471d-8092-52a22e0920bc.png)

- Pedestrian Signal example 
![image](https://user-images.githubusercontent.com/106000281/234416681-3039e6de-5024-42c5-9e63-10910f166329.png)


##### Testing instructions
1. Navigate to Explore page
2. Place the first label (or click on existing label) and check context menu connection
3. Close and reopen context menu

##### Things to check before submitting the PR
